src/frontend/src/optimizer/mod.rs

```diff
@@ -376,6 +376,12 @@ impl PlanRoot {
 
         let plan = match self.plan.convention() {
             Convention::Logical => {
+                if let Some(err) = StreamKeyChecker.visit(self.plan.clone()) {
+                    return Err(ErrorCode::NotSupported(
+                        err,
+                        "Use a large Jsonb as part of the stream key is unexpected. If you are sure your Jsonb is small, use `set XXX true`".to_string(),
+                    ).into());
+                }
```

Comment:
I think we should check the stream key at least after the logical optimization phase because after column pruning, some unnecessary columns check could be skipped.

Reply:
The plan nodes after optimization can be different from the plan after binding and cause inaccurate error messages.

The columns checked are `group by` `order by` `partition by`, which is unlikely to be pruned.

Reply:
But for union, it seems hard to say? e.g. `select a, b, c from ( select * from t1 union all select * from t2)`

Reply:
you mean `union` here ? actually I am not sure  about the behavior here... can we prune the column here? in other words, I consider they are not equalevant
```SQL
SELECT a,b FROM (
    SELECT distinct a,b,c FROM t
) t;

SELECT distinct a,b FROM FROM t;
```

Reply:
fixed an issue about union all

Reply:
Agg can be created during subquery unnesting. See the sample query here: https://github.com/risingwavelabs/risingwave/issues/7981#issuecomment-1882631969

Btw is join key a concern as well?

Reply:
Yes, this or only handles user explicitly write agg queries. We need a fallback to handle other cases.

Yes, join can be a concern.